Maximum recursion level exceeded

The latest version (0.8.0) has introduced a pretty significant issue that eventually leads to the following stack trace or something similar:

```
2015-10-08 14:29:59.008 - [ERROR] (12366) (MainProcess)(MainThread) --> Caught unhandled exception: maximum recursion depth exceeded
Traceback (most recent call last):
  File "/someapp/env/lib/python2.7/site-packages/flask/app.py", line 1475, in full_dispatch_request
    rv = self.dispatch_request()
  File "/someapp/env/lib/python2.7/site-packages/flask/app.py", line 1461, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/someapp/env/lib/python2.7/site-packages/flask_restful/__init__.py", line 471, in wrapper
    resp = resource(*args, **kwargs)
  File "/someapp/env/lib/python2.7/site-packages/flask/views.py", line 84, in view
    return self.dispatch_request(*args, **kwargs)
  File "/someapp/env/lib/python2.7/site-packages/flask_restful/__init__.py", line 581, in dispatch_request
    resp = meth(*args, **kwargs)
  File "/someapp/env/lib/python2.7/site-packages/flask_restplus/resource.py", line 27, in wrapper
    return func(*args, **kwargs)
  File "/someapp/env/lib/python2.7/site-packages/flask_restplus/resource.py", line 27, in wrapper
    return func(*args, **kwargs)
  File "/someapp/env/lib/python2.7/site-packages/flask_restplus/resource.py", line 27, in wrapper
    return func(*args, **kwargs)
  ...
  File "/someapp/env/lib/python2.7/site-packages/flask_restplus/resource.py", line 27, in wrapper
    return func(*args, **kwargs)
  File "/someapp/env/lib/python2.7/site-packages/flask_restplus/resource.py", line 20, in wrapper
    data = request.get_json()
  File "/someapp/env/lib/python2.7/site-packages/werkzeug/local.py", line 338, in __getattr__
    return getattr(self._get_current_object(), name)
  File "/someapp/env/lib/python2.7/site-packages/werkzeug/local.py", line 297, in _get_current_object
    return self.__local()
  File "/someapp/env/lib/python2.7/site-packages/flask/globals.py", line 18, in _lookup_req_object
    top = _request_ctx_stack.top
  File "/someapp/env/lib/python2.7/site-packages/werkzeug/local.py", line 163, in top
    return self._local.stack[-1]
RuntimeError: maximum recursion depth exceeded
```

Traced the problem to the following lines [here](https://github.com/noirbizarre/flask-restplus/blob/master/flask_restplus/resource.py#L13-L14):

```
class Resource(restful.Resource):

    def __init__(self, api, *args, **kwargs):
        self.api = api
        if self.validate_payload not in self.method_decorators:
            self.method_decorators.insert(0, self.validate_payload)
```

The `if self.validate_payload not in self.method_decorators:` always returns true and just keeps adding the decorator over and over because `method_decorators` is a class level property declared as a `list` (mutable python object problem) and `Resource` is instantiated as a new object each time.

So when running that list changes the following ways:

```
[]
127.0.0.1 - - [08/Oct/2015 14:35:51] "GET /api-aw/jobs HTTP/1.0" 200 -
[<bound method JobsList.validate_payload of <app.api.resources.someapp.jobs.JobsList object at 0x1fccc90>>]
127.0.0.1 - - [08/Oct/2015 14:35:51] "GET /api-aw/jobs HTTP/1.0" 200 -
[<bound method JobsList.validate_payload of <app.api.resources.someapp.jobs.JobsList object at 0x1fccb50>>, <bound method JobsList.validate_payload of <app.api.resources.someapp.jobs.JobsList object at 0x1fccc90>>]
127.0.0.1 - - [08/Oct/2015 14:35:51] "GET /api-aw/jobs HTTP/1.0" 200 -
[<bound method JobsList.validate_payload of <app.api.resources.someapp.jobs.JobsList object at 0x1fccb10>>, <bound method JobsList.validate_payload of <app.api.resources.someapp.jobs.JobsList object at 0x1fccb50>>, <bound method JobsList.validate_payload of <app.api.resources.someapp.jobs.JobsList object at 0x1fccc90>>]
127.0.0.1 - - [08/Oct/2015 14:35:51] "GET /api-aw/jobs HTTP/1.0" 200 -
[<bound method JobsList.validate_payload of <app.api.resources.someapp.jobs.JobsList object at 0x2147fd0>>, <bound method JobsList.validate_payload of <app.api.resources.someapp.jobs.JobsList object at 0x1fccb10>>, <bound method JobsList.validate_payload of <app.api.resources.someapp.jobs.JobsList object at 0x1fccb50>>, <bound method JobsList.validate_payload of <app.api.resources.someapp.jobs.JobsList object at 0x1fccc90>>]
```

This eventually leads to the maximum recursion problem. One option is to just do the following:

```
class Resource(restful.Resource):

    def __init__(self, api, *args, **kwargs):
        self.api = api
        self.method_decorators = [self.validate_payload]
```

This does mean though you can't do what is explained [here](http://flask-restful.readthedocs.org/en/latest/extending.html#resource-method-decorators) any longer.

Needless to say this is pretty significant issue because pretty soon anyone using this is going to start having some pretty major problems as soon as they get enough requests for this to manifest itself.

--- a/flask_restplus/api.py
+++ b/flask_restplus/api.py
@@ -396,6 +396,19 @@ class Api(restful.Api):
     def as_postman(self, urlvars=False, swagger=False):
         return PostmanCollectionV1(self, swagger=swagger).as_dict(urlvars=urlvars)
 
+    def validate_payload(self, func):
+        '''Perform a payload validation on expected model'''
+        def wrapper(*args, **kwargs):
+            if hasattr(func, '__apidoc__'):
+                model = func.__apidoc__.get('body')
+                validate = func.__apidoc__.get('validate', False)
+                if model and validate and hasattr(model, 'validate'):
+                    # TODO: proper content negotiation
+                    data = request.get_json()
+                    model.validate(data, self.refresolver)
+            return func(*args, **kwargs)
+        return wrapper
+
     @property
     def payload(self):
         return request.get_json()

--- a/flask_restplus/resource.py
+++ b/flask_restplus/resource.py
@@ -1,8 +1,6 @@
 # -*- coding: utf-8 -*-
 from __future__ import unicode_literals
 
-from flask import request
-
 from flask.ext import restful
 
 
@@ -10,18 +8,8 @@ class Resource(restful.Resource):
 
     def __init__(self, api, *args, **kwargs):
         self.api = api
-        if self.validate_payload not in self.method_decorators:
-            self.method_decorators.insert(0, self.validate_payload)
+        self._add_validate_payload()
 
-    def validate_payload(self, func):
-        '''Perform a payload validation on expected model'''
-        def wrapper(*args, **kwargs):
-            if hasattr(func, '__apidoc__'):
-                model = func.__apidoc__.get('body')
-                validate = func.__apidoc__.get('validate', False)
-                if model and validate and hasattr(model, 'validate'):
-                    # TODO: proper content negociation
-                    data = request.get_json()
-                    model.validate(data, self.api.refresolver)
-            return func(*args, **kwargs)
-        return wrapper
+    def _add_validate_payload(self):
+        if self.api.validate_payload not in self.method_decorators:
+            self.method_decorators.insert(0, self.api.validate_payload)

